### ISSUE TITLE
 Unable to Build Project with c# solution (Godot 4.1.1)

### ISSUE THREAD

**Issue by ArtemSin** [godotengine/godot#81790]
### Godot version

4.1.1 Stable 

### System information

Windows 10 - Gotot 4.1.1 Stable 

### Issue description

Run Game failed if i have c# solution in project with these errors
 editor/editor_node.cpp:6468 - An EditorPlugin build callback failed.
  Transient parent has another exclusive child
 If i delete csproj file it runs well  , but cant rebuild ( it will regenerate csproj and have error again) 
I'm using Rider with gotot plugin ,MSBuild dotnet 6

### Steps to reproduce

Create Empty project
Create c# script
Create c# solution
Run it

### Minimal reproduction project

An empty project

**Comment by raulsntos** [comment]
Deleting the csproj will make Godot think your project doesn't contain C# code and will not build it.

In the steps to reproduce, you mention creating the solution as one of the steps after creating a script. Godot should automatically create the solution if it doesn't exist when creating a C# script, did that not happen for you?

This error is likely to show up when your C# project does not compile, check the MSBuild panel to see if there were any errors building your project.

Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/4c1f11944ebfe1204f21c6ac488c7a2294e1d520/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by ArtemSin** [comment]
@raulsntos yes, you right. If i create с#script - godot will make solution as well. In MSBuild panel there is one error- "The build method threw an exception." Actualy i dont have more info about that. There is reproduction project 
[Test5 copy.zip](https://github.com/godotengine/godot/files/12642714/Test5.copy.zip)

**Comment by raulsntos** [comment]
You should be able to right-click that error and copy it, it should contain the exception that was thrown. It could be that you don't have the dotnet CLI in the PATH environment variable, can you run the `dotnet --info` command in a console?

**Comment by ArtemSin** [comment]
@raulsntos i tried to copy error but it didnt provide more info , only "The build method threw an exception." There is dotnet info comand's result 
![image](https://github.com/godotengine/godot/assets/35371386/ec89bc2b-3e9a-425b-9192-c3109f05c323)

**Comment by ArtemSin** [comment]
@raulsntos i see, 4 real environment variables not set,  will try to fix this

**Comment by raulsntos** [comment]
> i tried to copy error but it didnt provide more info

That's weird, from what I can see that message should always include the exception in a new line:

https://github.com/godotengine/godot/blob/4df80b0e629e25653a706f6721c13a1f9d1da368/modules/mono/editor/GodotTools/GodotTools/Build/BuildManager.cs#L99-L101

Can you see the exception message if you hover the error? Also, since the exception is printed to the console, try opening Godot from a terminal and see the output there.

> i see, 4 real environment variables not set, will try to fix this

It seems that you do have dotnet in the PATH, otherwise you wouldn't be able to execute the command, right?

**Comment by ArtemSin** [comment]
@raulsntos  i tried to launch with console too, but there's no second line  . I checked and add again dotnet in the PATH ( maybe it already was), also i added DOTNET_ROOT variable too , but there's no effects 
![image](https://github.com/godotengine/godot/assets/35371386/a712e3b2-170c-4b3f-923b-625a25a11a78)

**Comment by ArtemSin** [comment]
@raulsntos in godot i changed verbosity level to detailed , so for now in console i see new exception 
![image](https://github.com/godotengine/godot/assets/35371386/be63cc82-2477-46d5-b1d9-ca4575c087a4)

**Comment by raulsntos** [comment]
It seems Godot can't find dotnet in your PATH environment variable. Try executing `echo %PATH%` in the terminal you are using to open Godot, it should contain the path to the directory that contains the dotnet executable.

**Comment by lukiiko** [comment]
I'm experiencing the same issue on Ubuntu 22.04 and I just verified that dotnet is in my path. The issue still persists. Also I was building perfectly fine with a C# solution. I seem to have done something however which made it no longer work. Since I just started this project last night, it's very small (like 4 scenes and 3 scripts). If it helped, I could upload it.

**Comment by lukiiko** [comment]
I have solved the issue on my end. I decided to run `dotnet build` in my project directory and it gave me a usable error message. It turns out I had a signal declared as private. After fixing that it builds normally again.

**Comment by ArtemSin** [comment]
@raulsntos  yeah, it contains .  I have some dotnet instances in diffrent paths , and also in system and user variables for sure. I tried to remove from  PATH  some other instances and leave just one (   from user documents /  program files ) , but there's no result. There is no a way to  set the path directly in godot  ?
![image](https://github.com/godotengine/godot/assets/35371386/d429686d-8e80-4338-8401-f23ae3deecc7)

![image](https://github.com/godotengine/godot/assets/35371386/22829280-f9eb-4a9a-8d02-9084d1b358fe)

**Comment by ArtemSin** [comment]
@raulsntos so , i was wrong about v3.5 , it can run c# code, but it also give me an error when i press Build button . Build tool is dotnet CLI 
![image](https://github.com/godotengine/godot/assets/35371386/b0e7ccb9-b25c-4b9e-8c74-2724d10f6862)

![image](https://github.com/godotengine/godot/assets/35371386/6ca0991f-ba9c-48c6-9512-2d2fe1ff233c)
![image](https://github.com/godotengine/godot/assets/35371386/b6f94d18-0d40-426c-a780-ea461b54da33)

**Comment by ArtemSin** [comment]
@raulsntos alright, i've found temporary solution from this topic ( about same problem on Mac)  https://github.com/godotengine/godot/issues/38985 
If i launch Godot from cmd , it works fine ,both 3.5 and 4.1 , so i can build project and run it without errors. It is wierd , but for now this is only solution for me

**Comment by cyraid** [comment]
I'm not sure if this will help anyone, but make sure in `project.godot`:
```
[dotnet]

project/assembly_name="NameOfSolution"
```

Make sure NameOfSolution.sln and NameOfSolution.csproj exists.

**Comment by HexyTulip** [comment]
@cyraid and @ArtemSin  can you please explain more about the possible causes of this? im on windows and on version 4.4.1 mono stable and i cant run any projects with c# scripts on it. there are no errors, the editor freezes on the loading bar of building .net project for hours with no change. it doesn't even use any CPU in task manager. none of the solutions previously mentioned seems to work for me. just to be clear, version 4.4 mono stable has no such issue and works normally.

![Image](https://github.com/user-attachments/assets/2b424804-7e0b-4d99-9a15-60729d104abb)

**Comment by IvanMurzak** [comment]
We hit this reliably as a headless-CI crash and reduced it to a minimal repro — sharing in case it helps triage:

**Symptom:** On Godot 4.3.0 / 4.4.1 / 4.5.1 (mono), a cold single-process `godot --headless --path <project> --build-solutions --quit` crashes with **SIGSEGV (exit 139)** when a C# `[Tool] EditorPlugin` is enabled in the project — right after `loading_editor_layout DONE`, **before any managed plugin code runs**. A trivial ~10-line zero-dependency `[Tool] EditorPlugin` crashes identically; durable breadcrumbs placed at `[ModuleInitializer]`, the plugin ctor, and `_EnterTree` are never reached.

**Works fine via separate processes:** `--headless --import --quit` → `dotnet build <project>.csproj` → `--headless --editor --quit`. So it is specific to the cold single-process `--build-solutions` path instantiating the first C# tool plugin.

**Public repro (both paths in CI):** https://github.com/IvanMurzak/Godot-AI-Particles — its `e2e-godot` workflow had to switch off the crashing cold path onto the 3-process pattern.
